### ISSUE TITLE
reuse of pyDot11

### ISSUE THREAD

**Issue by proocho** [ICSec/pyDot11#29]
Hello stryngs,

I am a IT student and I found your solution for decryption/encryption of WEP/WPA.

Could you please give me answer to these questions (as I do not know how to contact you):
* Is it okay if I use this in my thesis?
* Can I decrypt/encrypt WPA2 CCMP wifi frames?
* How do you generate PSK from SSID and password?

Thank you for your answer in advance.

**Comment by proocho** [comment]
Could you please give me an example of using wpaDecrypt function?

**Comment by stryngs** [comment]
Hi @proocho 

You're more than welcome to use this work in your thesis.  I've been taking care of some family matters, thus the response delay.

I'll see what I can't do, to make sometime tonight to put something together for you.

Cheers.

**Comment by proocho** [comment]
Hello @stryngs,

I fully understand that family comes first :). I am glad that you wrote me back. 

I have been studying your library closer and I found out that you creating the key from EAPOL packet sequence in **handshake.py**. Then later on it comes to decryption of packets in **ccmp.py**.

I am working on a component that only encrypts and decrypts WiFi traffic in SDWN (Software defined WiFi networks), so Access Points in the SDN topology works more effectively. That being said, they only sniff packet and forward it to this component and go to sniffing again. They do not need to decrypt packets to the DS or encrypt it from DS to the stations.

I found your library as the only solution fit for me, because I needed the decryption and encryption tool that can work on air. The other things I need to do (OpenFlow compatibility; forwarding rules; storing Stations and Access Points in some kind of forwarding table; test the solution; measure efficiency gains) are the main goal of my thesis.

So, if I could give you points that your library would be very helpful for me:
* Raw WiFi packet comes from AP, I need to decrypt it via WPA2 (aes ccmp) algorithm and send it via ethernet as Ethernet II packet.
* Ethernet II packet comes from ethernet interface, I need to encrypt it via WPA2 (aes ccmp) algorithm and send it via WiFi.

Thank you very much for your answer in advance.
